sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -440,6 +452,83 @@ case class GpuArrayMin(child: Expression) extends GpuUnaryExpression with Implic
     input.getBase.listReduce(SegmentedReductionAggregation.min())
 }
 
+/** ArrayMin without `Nan` handling */
+case class GpuBasicArrayMin(child: Expression) extends GpuArrayMin(child)
+
+/** ArrayMin for FloatType and DoubleType to handle `Nan`s.
+ *
+ * In Spark, `Nan` is the max float value, however in cuDF, the calculation
+ * involving `Nan` is undefined.
+ * We design a workaround method here to match the Spark's behaviour.
+ * The high level idea is:
+ *   if one list contains only `Nan`s or `null`s
+ *   then
+       if the list contains `Nan`
+ *     then return `Nan`
+ *     else return null
+ *   else
+ *     replace all `Nan`s with nulls;
+ *     use cuDF kernel to find the min value
+ */
+case class GpuFloatArrayMin(child: Expression) extends GpuArrayMin(child) {
+    @transient override lazy val dataType: DataType = child.dataType match {
+    case ArrayType(FloatType, _) => FloatType
+    case ArrayType(DoubleType, _) => DoubleType
+    case _ => throw new IllegalStateException(
+      s"GpuFloatArrayMin accepts only float array and double array."
+    )
+  }
+
+  protected def getNanSalar: Scalar = dataType match {
+    case FloatType => Scalar.fromFloat(Float.NaN)
+    case DoubleType => Scalar.fromDouble(Double.NaN)
+    case t => throw new IllegalStateException(s"dataType $t is not FloatType or DoubleType")
+  }
+
+  protected def getNullScalar: Scalar = dataType match {
+    case FloatType => Scalar.fromNull(DType.FLOAT32)
+    case DoubleType => Scalar.fromNull(DType.FLOAT64)
+    case t => throw new IllegalStateException(s"dataType $t is not FloatType or DoubleType")
+  }
+
+  override protected def doColumnar(input: GpuColumnVector): cudf.ColumnVector = {
+    val listAll = SegmentedReductionAggregation.min()
+    val listAny = SegmentedReductionAggregation.max()
+    val base = input.getBase()
+
+    withResource(base.getChildColumnView(0)) { child =>
+      withResource(child.isNan()){ childIsNan =>
+        // if all values in each list are nans or nulls
+        val allNanOrNull = {
+          val childIsNanOrNull = withResource(child.isNull()) {_.or(childIsNan)}
```

Comment:
Hmm, based on the doc of `isNan()`: https://github.com/rapidsai/cudf/blob/branch-22.10/java/src/main/java/ai/rapids/cudf/ColumnView.java#L463-L470
In the column `ChildIsNan`, all `null`s will be mapped to `False`. So that `SegmentedReductionAggregation.all()` cannot always give `True` if a list only contains `Nan` and `null`.